autoversioner/cmd/versiongenerator/versiongenerator.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	ghub "github.com/GoogleCloudPlatform/guest-test-infra/autoversioner/github"
+	"github.com/google/go-github/github"
+)
+
+var (
+	tokenFile = flag.String("token-file-path", "", "path to github token file")
+	org       = flag.String("org", "", "organization name")
+	repo      = flag.String("repo", "", "repository name")
+	sha       = flag.String("sha", "", "sha of commit id")
+)
+
+const (
+	githubDateFormat = "20060102"
+)
+
+func main() {
+	ctx := context.Background()
+	flag.Parse()
+	err := validateFlags()
+	if err != nil {
+		fmt.Printf("error validating flags: %+v", err)
+		os.Exit(1)
+	}
+
+	buildID, err := generateVersion(ctx)
+	if err != nil {
+		fmt.Printf("error generating build id: %+v", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Generated version: %s\n", buildID)
+}
+
+// generateVersion generates the version string.
+// The algo to generate it is as follows:
+// - for the given sha; fetch the date on which it was committed
+// - fetch all the commits for that date,
```

Comment:
should only be valid for today, no need to support historical version building. this can be much simplified by asking for the latest

Reply:
same as the other comment